poetry/utils/setup_reader.py

```diff
@@ -1,379 +1,96 @@
-import ast
+import re
+import shutil
 
-from typing import Any
-from typing import Dict
-from typing import Iterable
-from typing import List
-from typing import Optional
-from typing import Tuple
-from typing import Union
+import pep517.envbuild
+import pep517.wrappers
+
+from poetry.utils.helpers import temporary_directory
 
-from ._compat import PY35
 from ._compat import Path
-from ._compat import basestring
 
 
 try:
-    from configparser import ConfigParser
+    from importlib.metadata import PathDistribution
 except ImportError:
-    from ConfigParser import ConfigParser
+    from importlib_metadata import PathDistribution
 
 
 class SetupReader(object):
     """
     Class that reads a setup.py file without executing it.
     """
 
-    DEFAULT = {
-        "name": None,
-        "version": None,
-        "install_requires": [],
-        "extras_require": {},
-        "python_requires": None,
-    }
-
-    FILES = ["setup.py", "setup.cfg"]
+    build_backend = "setuptools.build_meta"
+    build_requires = "setuptools"
 
     @classmethod
-    def read_from_directory(
-        cls, directory
-    ):  # type: (Union[basestring, Path]) -> Dict[str, Union[List, Dict]]
-        if isinstance(directory, basestring):
-            directory = Path(directory)
-
-        result = cls.DEFAULT.copy()
-        for filename in cls.FILES:
-            filepath = directory / filename
-            if not filepath.exists():
-                continue
-
-            new_result = getattr(cls(), "read_{}".format(filename.replace(".", "_")))(
-                filepath
-            )
-
-            for key in result.keys():
-                if new_result[key]:
-                    result[key] = new_result[key]
-
-        return result
-
-    @classmethod
-    def _is_empty_result(cls, result):  # type: (Dict[str, Any]) -> bool
-        return (
-            not result["install_requires"]
-            and not result["extras_require"]
-            and not result["python_requires"]
+    def read_from_pep517_hook(cls, directory):
```

Comment:
Fwiw, I think it might be great to retain the name for now. Since we are reading form the givent directory.